File: 프로그래머스/0/181894. 2의 영역/2의 영역.py
def solution(arr):
    s,e=-1,-1
    for i in range(len(arr)):
        if arr[i]==2:
            s=i
            break
    for i in range(len(arr)-1,-1,-1):
        if arr[i]==2:
            e=i
            break
    answer=[-1] if s==-1 and e==-1 else arr[s:e+1]
    return answer

File: 프로그래머스/0/181894. 2의 영역/2의 영역.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> r;
    int st=0, ed=0;
    for(int i=arr.size()-1;i>0;i--) {
        if(arr[i]==2) {
            st=find(arr.begin(),arr.end(),2)-arr.begin();
            ed=i;
            break;
        }   
    }
    if(st==0 and ed==0) {
        r.push_back(-1);
    } else {
        for(int i=st;i<ed+1;i++) {
            r.push_back(arr[i]);
        }
    }
    return r;
}